app/screens/thread/thread_post_list/thread_post_list.tsx

```diff
@@ -31,18 +35,28 @@ const styles = StyleSheet.create({
 });
 
 const ThreadPostList = ({
-    channelId, currentTimezone, currentUsername,
-    isTimezoneEnabled, lastViewedAt, nativeID, posts, rootPost,
+    channel, currentTimezone, currentUsername,
+    isCRTEnabled, isTimezoneEnabled, lastViewedAt, nativeID, posts, rootPost,
 }: Props) => {
     const isTablet = useIsTablet();
+    const serverUrl = useServerUrl();
 
     const threadPosts = useMemo(() => {
         return [...posts, rootPost];
     }, [posts, rootPost]);
 
+    // If CRT is enabled, When new post arrives and thread modal is open, mark thread as read
+    const oldPostsCount = useRef<number>(posts.length);
+    useEffect(() => {
+        if (isCRTEnabled && oldPostsCount.current < posts.length) {
+            oldPostsCount.current = posts.length;
+            updateThreadRead(serverUrl, channel.teamId, rootPost.id, Date.now());
+        }
+    }, [channel, isCRTEnabled, posts, rootPost, serverUrl]);
```

Comment:
Why is this not handled in the WS event ?

Reply:
Because we don't have a way to know if the thread screen is open or not.